addons/LPCAnimatedSprite/LPCSpriteData.gd

```diff
@@ -5,9 +5,15 @@ var Name:String
 var Row:int
 var Col:int
 var Loop:bool
-func _init(frameCount:int, name:String, row:int, col:int, loop:bool):
+var Direction:Vector2
+var Type:String
+var Oversize:bool
+func _init(frameCount:int, name:String, row:int, col:int, loop:bool, direction:Vector2, type: String, oversize: bool):
 		FrameCount = frameCount
 		Name = name
 		Row = row
 		Col = col
 		Loop = loop
+		Direction = direction
+		Type = type
+		Oversize = oversize
```

Comment:
If you accept this change [here](https://github.com/alextrevisan/LPCAnimatedSprite2D/pull/22/files#r1428993588) we do not need these extras data here.

Don't botter with the oversized, I'm reworking on it for now

Reply:
I think you're probably right about this. Let me sleep on it, but I think I'm sold.

Would it be reasonable to convert LPCAnimationData into a dictionary (and use get_set functions to give structure rather than the LPCSpriteData class) instead?

Should be faster for other sides of the implementation as well.

Reply:
> I think you're probably right about this. Let me sleep on it, but I think I'm sold.
> 
> Would it be reasonable to convert LPCAnimationData into a dictionary (and use get_set functions to give structure rather than the LPCSpriteData class) at the same time?
> 
> Should be faster for other sides of the implementation as well.

I don't think I understood, can you show me a snippet of the code that you want?

Reply:
![image](https://github.com/alextrevisan/LPCAnimatedSprite2D/assets/5817861/d0439e66-a8a9-412f-bf25-0f004009d4fd)

could become something like this instead:

![Screenshot_20231216_224714](https://github.com/alextrevisan/LPCAnimatedSprite2D/assets/5817861/2b572a5c-6250-46bb-b68a-f048313a968f)

just a thought.

Reply:
This is not how dictionary works. When you want to add a value to a dictionary, it needs a specific key. If you do:
AnimationData[name][framecount] = framecount

and in another line you do
AnimationData[name][row] = row

if row and framecount are the number 7, you are telling the dictionary to access:

AnimationData["WALK"][7] = 7

And if you want to retrieve how many frames this animation have, you will need to know the key (in this case 7)

But the acess time for a dictionary is way slower than a class with specifc atributes, like is today.

The dictionary is good to store relational data like an array, but with easy indexing

Reply:
😅 Right, right, forgot to put the second index in quotes, when I start making errors like this it's a good sign it's time for bed.

I see your broader point though and concede it. I appreciate you indulging me in these questions, for what it's worth. Thank you!